src/components/UserPage/UserPage.js

```diff
@@ -1,10 +1,135 @@
-import React, { Component } from 'react';
+import React, { Component, useEffect } from 'react';
 import { connect } from 'react-redux';
 import LogOutButton from '../LogOutButton/LogOutButton';
 import mapStoreToProps from '../../redux/mapStoreToProps';
 
+const mapStateToProps = (state, ownProps) => ({
+  dispatch: ownProps.dispatch,
+  history: ownProps.history,
+  user: state.user,
+  friends: state.friendsReducer,
+  decks: state.deck,
+})
+
+const UserPageFunc = ({
+  dispatch,
+  history,
+  user,
+  friends,
+  decks,
+}) => {
+  useEffect(() => {
+    dispatch({ type: 'FETCH_USER' });
+    dispatch({ type: 'GET_DECK' });
+    dispatch({ type: 'GET_FRIENDS' });
+  }, [dispatch])
+
+  const createDeck = () => {
+    const deck = prompt("Please enter a deck name:", "My Awesome Deck");
+    if (deck === null || deck === "") {
+      alert("Canceled");
+    } else {
+      this.props.dispatch({
+        type: 'CREATE_DECK',
+        payload: {
+          userid: user.id,
+          deckname: deck,
+          ispublic: false,
+          description: "",
+          decklist: '',
+          featured_card: '',
+          upvotes: 0,
+          comments: '',
+        },
+      });
+    };
+  };
+
+  const deleteDeck = (deck) => {
+    const response = window.confirm(`Are you sure you want to permanenently delete this deck?`)
+    if (response){
+      this.props.dispatch({
+        type: 'DELETE_DECK',
+        payload: deck
+      })
+    }
+  };
+
+  return (
+    <div id='userMainDiv'>
+      <img id='profilePic' height='130px' width='130px' src={user.img_url} />
+      <div id='profileInfo'>
+        <h1 id='welcome'>
+          Welcome, {user.username}!
+        </h1>
+        <button onClick={() => history.push('/edituser')}>
+          Edit Profile
+        </button>
+      </div>
+      <hr />
+      <div id='mainDiv'>
+        <div id='userDiv'>
+          <h1>
+            {user.username}'s Decks
+          </h1>
+          <br/>
+          <button onClick={createDeck}>
+            Create New Deck
+          </button>
+          <hr/>
+          <div id='userDeckScroll'>
+            {decks.map((deck) =>
+              <div id='deckOptions' key={deck.id}>
```

Comment:
Way too many ids. The article I shared will help, but ids should aways only exist once. They are meant to be a unique identifier used for one single rendered tag in your application. If multiple rendered tags have the same id that is a problem.
Since each tag can only have one id, you also have more divs than necessary here. classes will allow you to apply multiple to the same div so you don't need so many nested.